src/overlays/actors/ovl_En_Po_Sisters/z_en_po_sisters.c

```diff
@@ -85,114 +141,1056 @@ static DamageTable D_80B1DAA4 = {
+        (this->actionFunc != func_80B1B444)) {
+        if (this->unk_18F == 0) {
+            func_80B1B70C(this);
+        } else if ((this->unk_194 == 0) && (this->unk_229 == 0)) {
+            func_80B1B860(this, globalCtx);
+        }
+    }
+}
+
+void func_80B1AA88(EnPoSisters* this) {
+    SkelAnime_ChangeAnimTransitionRepeat(&this->skelAnime, &D_060014CC, -3.0f);
+    this->actor.speedXZ = 0.0f;
+    this->unk_192 = Rand_S16Offset(100, 50);
+    this->actionFunc = func_80B1AAE8;
+}
+
+void func_80B1AAE8(EnPoSisters* this, GlobalContext* globalCtx) {
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    if (DECR(this->unk_192) == 0) {
+        this->unk_192 = Rand_S16Offset(100, 50);
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_PO_LAUGH2);
+    }
+}
+
+void func_80B1AB5C(EnPoSisters* this) {
+    SkelAnime_ChangeAnimTransitionRepeat(&this->skelAnime, &D_060014CC, -3.0f);
+    this->unk_192 = Rand_S16Offset(2, 3);
+    this->actor.speedXZ = 0.0f;
+    this->actionFunc = func_80B1ABB8;
+}
+
+void func_80B1ABB8(EnPoSisters* this, GlobalContext* globalCtx) {
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    if (func_801378B8(&this->skelAnime, 0.0f)) {
+        if (this->unk_192 != 0) {
+            this->unk_192--;
+        }
+    }
+
+    if ((this->unk_192 == 0) || (this->actor.xzDistToPlayer < 600.0f)) {
+        func_80B1AC40(this);
+    }
+}
+
+void func_80B1AC40(EnPoSisters* this) {
+    if (this->actionFunc != func_80B1AE3C) {
+        SkelAnime_ChangeAnimTransitionRepeat(&this->skelAnime, &D_06000D40, -3.0f);
+    }
+    this->unk_192 = Rand_S16Offset(15, 3);
+    this->unk_191 |= (0x6 | 0x1);
+    this->actionFunc = func_80B1ACB8;
+}
+
+void func_80B1ACB8(EnPoSisters* this, GlobalContext* globalCtx) {
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    Math_StepToF(&this->actor.speedXZ, 1.0f, 0.2f);
+    if (func_801378B8(&this->skelAnime, 0.0f)) {
+        if (this->unk_192 != 0) {
+            this->unk_192--;
+        }
+    }
+
+    if ((this->actor.xzDistToPlayer < 600.0f) && (fabsf(this->actor.yDistToPlayer + 5.0f) < 30.0f)) {
+        func_80B1AE28(this);
+    } else if ((this->unk_192 == 0) && Math_StepToF(&this->actor.speedXZ, 0.0f, 0.2f)) {
+        func_80B1AB5C(this);
+    }
+
+    if (this->actor.bgCheckFlags & 8) {
+        Math_ScaledStepToS(&this->actor.world.rot.y, Actor_YawToPoint(&this->actor, &this->actor.home.pos), 0x71C);
+    } else if (Actor_XZDistanceToPoint(&this->actor, &this->actor.home.pos) > 600.0f) {
+        Math_ScaledStepToS(&this->actor.world.rot.y, Actor_YawToPoint(&this->actor, &this->actor.home.pos), 0x71C);
+    }
+}
+
+void func_80B1AE28(EnPoSisters* this) {
+    this->actionFunc = func_80B1AE3C;
+}
+
+void func_80B1AE3C(EnPoSisters* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    s16 sp22;
+
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    sp22 = this->actor.yawTowardsPlayer - player->actor.shape.rot.y;
```

Comment:
```suggestion
    s16 sp22 = this->actor.yawTowardsPlayer - player->actor.shape.rot.y;

    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
```
Does this work, or does it mess up the stack?